com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/eval/JdtEvaluationProvider.java

```diff
@@ -132,6 +135,37 @@ protected synchronized void initialize() {
         return completableFuture;
     }
 
+    private IJavaProject calculateJavaProject(String projectName, ThreadReference thread, int depth) {
+        if (StringUtils.isBlank(projectName)) {
+            try {
+                StackFrame sf = thread.frame(depth);
+                String typeName = sf.location().method().declaringType().name();
+
+                // check whether the project is the only java project in workspace
+                IWorkspaceRoot root = ResourcesPlugin.getWorkspace().getRoot();
+                List<IJavaProject> projects = Arrays.stream(root.getProjects()).map(JdtUtils::getJavaProject).filter(p -> {
+                    try {
+                        return p.findType(typeName) != null && p.hasBuildState();
```

Comment:
The findType should be heavy API. If there is only one project, we can safely bet the projectName?